### ISSUE TITLE
Zero amount verification in ThreeDSecure process

### ISSUE THREAD

**Issue by puniserv** [braintree/braintree-web#620]
### General information

* SDK version: 3.85.2 and older
* Environment: every environment
* Browser and OS: Chrome 99.0.4844.51 (64-bit) on MacOS 12.2.1

### Issue description

Zero amount verification throws error when numeric type is used:
```
.verifyCard({
        amount: 0,
        ...
 }, ...)
  ```

String type is working, but is not compatible with types/braintree-web definitions.
```
.verifyCard({
        amount: '0',
        ...
 }, ...)
  ```

Error:
```
{name: 'BraintreeError', code: 'THREEDS_MISSING_VERIFY_CARD_OPTION', message: 'verifyCard options must include an amount.', type: 'MERCHANT', details: undefined}
```

Problem is in `braintree-web/three-d-secure/external/frameworks/base.js` file:
Probably `if(!options.amount)` should be changed to null|undefined === condition .

```
  if (this._verifyCardInProgress === true) {
    return new BraintreeError(errors.THREEDS_AUTHENTICATION_IN_PROGRESS);
  } else if (!options.nonce) {
    errorOption = 'a nonce';
  } else if (!options.amount) {
    errorOption = 'an amount';
  }
```

**Comment by puniserv** [comment]
@jplukarski not really. You can process card verification with 0 amount. We are using this when client is adding card to our system and 0 amount does not cause `94505: The amount you specified must be a number greater than 0.` error

**Comment by JoshStrobl** [comment]
Ran into this as well and had to use `yarn patch` to remove the option checks. Would be better if instead of inverted truthy value (where 0 is false), it just checked that is of type Number.
